lib/efolder/service.rb

```diff
@@ -57,7 +57,7 @@ def bgs_doc_uuids
             uuids << uploaded_docs[:uuid_txt]
           else
             uploaded_docs.each do |doc|
-              uuids << doc[:uuid_txt]
+              uuids << doc[:uuid_txt] if doc.is_a?(Hash)
```

Comment:
Should you also check to see if uuid_txt is a key?
```suggestion
              uuids << doc[:uuid_txt] if doc.is_a?(Hash) && doc.key?(:uuid_txt)
```

Reply:
If so, this should be added to the line above

Reply:
Do we know why it's not a Hash?

```ruby
Rails.logger.debug "Unexpected doc type: #{doc.class}" unless doc.is_a?(Hash)
```

If it is an array, do we need to iterate or grab the first one?

```ruby
uuids << doc[0][:uuid_txt] if doc.is_a?(Array) && doc.first.is_a?(Hash)
```